File: pjt_tournois_e_sport_boot/src/main/java/projet/sopra/pjt_tournois_e_sport_boot/model/Resultat.java
package projet.sopra.pjt_tournois_e_sport_boot.model;

import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinColumns;
import javax.persistence.ManyToOne;
import javax.persistence.SequenceGenerator;
import javax.persistence.Table;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name="Resultat")
@SequenceGenerator(name="seqResultat", sequenceName = "seq_resultat",initialValue=100,allocationSize = 1)
public class Resultat {
	@Id
	@Column(name="id")
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "seqResultat")
	@JsonView({Views.ResultatWithInscriptionAndMatch.class,Views.Match.class,Views.InscriptionWithId.class})
	private Long id;
	
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "resultat_match_id", foreignKey = @ForeignKey(name="resultat_match_fk"))
	@JsonView(Views.ResultatWithInscriptionAndMatch.class)
	@NotNull
	//@OnDelete( action = OnDeleteAction.CASCADE)
	private Match match;
	
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumns(
			{@JoinColumn(name = "resultat_participant_id", foreignKey = @ForeignKey(name="resultat_participant_fk")),
			@JoinColumn(name="resultat_tournoi_id",  foreignKey = @ForeignKey(name="resultat_tournoi_fk"))})
	@JsonView({Views.ResultatWithInscriptionAndMatch.class, Views.MatchWithIncriptions.class})
	//@OnDelete(action = OnDeleteAction.CASCADE)
	private Inscription participant;
	
	@Column(name="position_match")
	@JsonView({Views.ResultatWithInscriptionAndMatch.class, Views.MatchWithIncriptions.class})
	@NotNull
	private int positionMatch;
	
	@Column(name="score_match")
	@JsonView({Views.ResultatWithInscriptionAndMatch.class, Views.MatchWithIncriptions.class})
	@NotNull
	private int scoreMatch;

	
	public Resultat() {
	}

	public Resultat(Match match, Inscription participant, int positionMatch, int scoreMatch) {
		super();
		this.match = match;
		this.participant = participant;
		this.positionMatch = positionMatch;
		this.scoreMatch = scoreMatch;
	}

	
	public Long getId() {
		return id;
	}

	public Match getMatch() {
		return match;
	}

	public Inscription getParticipant() {
		return participant;
	}

	public int getPositionMatch() {
		return positionMatch;
	}

	public int getScoreMatch() {
		return scoreMatch;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public void setMatch(Match match) {
		this.match = match;
	}

	public void setParticipant(Inscription participant) {
		this.participant = participant;
	}

	public void setPositionMatch(int positionMatch) {
		this.positionMatch = positionMatch;
	}

	public void setScoreMatch(int scoreMatch) {
		this.scoreMatch = scoreMatch;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Resultat other = (Resultat) obj;
		return Objects.equals(id, other.id);
	}
	
	
	
	
}

File: pjt_tournois_e_sport_angular/src/app/model/resultat.ts
import { Match } from './match';
import { Inscription } from './inscription';

export class Resultat {
  _id: number | undefined;
  _match: Match | undefined;
  _participant: Inscription | undefined;
  _positionMatch: number | undefined;
  _scoreMatch: number | undefined;

  constructor(
    id?: number,
    match?: Match,
    participant?: Inscription,
    positionMatch?: number,
    scoreMatch?: number
  ) {
    this._id = id;
    this._match = match;
    this._participant = participant;
    this._positionMatch = positionMatch;
    this._scoreMatch = scoreMatch;
  }

  /**
   * Getter id
   * @return {number }
   */
  public get id(): number | undefined {
    return this._id;
  }

  /**
   * Getter match
   * @return {Match }
   */
  public get match(): Match | undefined {
    return this._match;
  }

  /**
   * Getter participant
   * @return {Inscription }
   */
  public get participant(): Inscription | undefined {
    return this._participant;
  }

  /**
   * Getter positionMatch
   * @return {number }
   */
  public get positionMatch(): number | undefined {
    return this._positionMatch;
  }

  /**
   * Getter scoreMatch
   * @return {number }
   */
  public get scoreMatch(): number | undefined {
    return this._scoreMatch;
  }

  /**
   * Setter id
   * @param {number } value
   */
  public set id(value: number | undefined) {
    this._id = value;
  }

  /**
   * Setter match
   * @param {Match } value
   */
  public set match(value: Match | undefined) {
    this._match = value;
  }

  /**
   * Setter participant
   * @param {Inscription } value
   */
  public set participant(value: Inscription | undefined) {
    this._participant = value;
  }

  /**
   * Setter positionMatch
   * @param {number } value
   */
  public set positionMatch(value: number | undefined) {
    this._positionMatch = value;
  }

  /**
   * Setter scoreMatch
   * @param {number } value
   */
  public set scoreMatch(value: number | undefined) {
    this._scoreMatch = value;
  }
}
